### PR TITLE
feat: implement SettlementSnapshotRepository persistence adapter

### PR DESCRIPTION
**Pull request by bjcoombs**
## Summary

Implements `domain.SettlementSnapshotRepository` persistence adapter for the reconciliation service (Task Master: reconciliation-service.14).

- **Create/CreateBatch**: Single and atomic batch insert (batch size 100) with tenant isolation via `withTenantTransaction()`
- **FindByID/FindByRunID**: Query by snapshot_id or run_id with `ErrNotFound` mapping
- **DeleteByRunID**: Idempotent cleanup for retry scenarios
- **MarkRunSnapshotsFinal**: Bulk JSONB merge to add `settlement_type=FINAL`, with CockroachDB-compatible NULL/json-null handling

Follows established patterns from `SettlementRunRepository` (PR #858) and `DisputeRepository`.

## Changes Made

| File | Description |
|------|-------------|
| `services/reconciliation/adapters/persistence/settlement_snapshot_repository.go` | Repository implementation with entity mapping |
| `services/reconciliation/adapters/persistence/settlement_snapshot_repository_test.go` | Integration tests with CockroachDB testcontainers |

## Technical Details

- **Entity mapping**: `SettlementSnapshotEntity` maps to `settlement_snapshot` table with `decimal(38,18)` for balance fields
- **JSONB merge**: `MarkRunSnapshotsFinal` uses `CASE WHEN` to handle both SQL NULL and JSON null before JSONB concatenation (CockroachDB `||` operator requires non-null JSONB objects)
- **Batch atomicity**: `CreateBatch` uses `tx.CreateInBatches(entities, 100)` within a single transaction for rollback guarantee

## Testing

8 integration tests with CockroachDB testcontainers:

| Test | Coverage |
|------|----------|
| TestCreate | Insert and verify persistence |
| TestCreateBatch | 250 snapshots across 3 batches |
| TestCreateBatchEmpty | Nil and empty slice edge cases |
| TestFindByID | Decimal precision (38,18) preservation + not found |
| TestFindByRunID | Multi-run filtering + empty result for unknown run |
| TestDeleteByRunID | Cleanup + idempotent re-delete |
| TestMarkRunSnapshotsFinal | JSONB merge with NULL, existing, and mixed attributes |
| TestCreateBatchRollback | Constraint violation mid-batch verifies atomic rollback |

### CODE REVIEW

File: services/reconciliation/adapters/persistence/settlement_snapshot_repository_test.go [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+	surrogateID := uuid.New()
+	runID := uuid.New()
+
+	err := db.WithContext(ctx).Exec(
+		fmt.Sprintf(`INSERT INTO %s."settlement_run" (id, run_id, account_id, scope, settlement_type, period_start, period_end, initiated_by) VALUES (?, ?, 'ACC-001', 'ACCOUNT', 'DAILY', NOW() - INTERVAL '1 day', NOW(), 'system')`, quoted),
+		surrogateID, runID,
+	).Error
+	require.NoError(t, err)
+	return surrogateID
+}
```
[the remark this answers is not in the document]
Comment (bjcoombs) [reply services/reconciliation/adapters/persistence/settlement_snapshot_repository_test.go:34]: Returning the surrogate `id` is intentional. The `settlement_snapshot.run_id` column is a FK to `settlement_run.id` (the surrogate PK), not to the business `run_id`. See the migration: `"run_id" uuid NOT NULL REFERENCES "settlement_run" ("id") ON DELETE CASCADE`. The test helper returns the surrogate ID because that is what the snapshot entity needs for the FK relationship.
